ystia/yorc
# Feature/gh 552 logs store

# Pull Request description
Refactor logs package to allow to use new implementations.

## Description of the change
### What I did

This PR contains https://github.com/ystia/yorc/pull/580.
And:

- Events package (only stuff about logs) refactoring
- Encryptor utility for encrypt/decrypt data
- New Store interface methods to handle logs:
  - GetLastModifyIndex
  - List

- Storage configuration management
It allows to configure stores of log, event and deployment type.
- Update documentation


### How to verify it
- Read the doc !
- Add in Yorc config file the lines:

```
  "storage": {
    "stores": [
      {
        "name": "myCipherFileStore",
        "implementation": "cipherFileCache",
        "types":  ["Log"],
        "properties": {
          "secret_key": "myverystrongpasswordo32bitlength"
        }
      }
    ]
  }
```

- Check correct config is saved in Consul
- Deploy an app with the related yorc a4c plugin https://github.com/alien4cloud/alien4cloud-yorc-provider/pull/15)
- Check logs are encrypted and stored in work/store directory.
- Check logs are correctly polled from A4C and Yorc CLI

### Description for the changelog
Refactor logs to allow to plug new implementations (GH-552](https://github.com/ystia/yorc/issues/552))
## Applicable Issues
#552 
#554
